e2e/test/scenarios/question/reproductions/42065-mongo-filter-by-id.cy.spec.js

```diff
@@ -0,0 +1,58 @@
+import { WRITABLE_DB_ID } from "e2e/support/cypress_data";
+import { restore, openTable } from "e2e/support/helpers";
+
+describe(
+  "issue 42010 -- Unable to filter by mongo id",
+  { tags: "@mongo" },
+  () => {
+    beforeEach(() => {
+      restore("mongo-5");
+      cy.signInAsAdmin();
+
+      cy.request(`/api/database/${WRITABLE_DB_ID}/schema/`).then(({ body }) => {
+        const tableId = body.find(table => table.name === "orders").id;
+        openTable({
+          database: WRITABLE_DB_ID,
+          table: tableId,
+          mode: "notebook",
+        });
+      });
+    });
+
+    it("should be possible to filter by Mongo _id column (metabase#42010)", () => {
+      cy.get("button").contains("Visualize").click();
+      cy.get("[data-testid=cell-data]")
+        .eq(10)
+        .then(elm => {
+          // Get the id of some row and go back to notebook editor. I suppose it differs per snapshot, hence we are
+          // unable to hardcode the value.
+          const id = elm.text();
+          cy.go("back");
+
+          // Check the notebook editor
+          cy.log("Check notebook editor");
+          cy.findByRole("button", { name: "Filter" }).click();
+          // Following selects first item from filter column picker
+          cy.get("[data-element-id=list-item-title]")
+            .eq(0)
+            .should("have.text", "ID");
+          cy.get("[data-element-id=list-item-title]").eq(0).click();
```

Comment:
You can (and should) chain the `.click()` directly on the previous selector. No need to "get it" again.

But more than that, we should scope this to the popover that appears.
```suggestion
          popover()
            .findAllByRole("option").first()
            .should("have.text", "ID")
            .click();
```